core/src/View/MDSpan/Kokkos_MDSpan_Layout.hpp

```diff
@@ -0,0 +1,120 @@
+/// extents
+template <class Extents>
+KOKKOS_INLINE_FUNCTION auto dimension_from_extent(const Extents &e,
+                                                  std::size_t r) noexcept {
+  return Extents::static_extent(r) == dynamic_extent ? e.extent(r) : 0;
+}
+
+template <class ArrayLayout, class MDSpanType>
+KOKKOS_INLINE_FUNCTION auto array_layout_from_mapping(
+    const typename MDSpanType::mapping_type &mapping) {
+  using mapping_type = typename MDSpanType::mapping_type;
+  using extents_type = typename mapping_type::extents_type;
+
+  constexpr auto rank = extents_type::rank();
+  const auto &ext     = mapping.extents();
+
+  static_assert(rank <= ARRAY_LAYOUT_MAX_RANK,
+                "Unsupported rank for mdspan (must be <= 8)");
+
+  if constexpr (std::is_same_v<ArrayLayout, LayoutStride>) {
+    return Kokkos::LayoutStride{
+        rank > 0 ? dimension_from_extent(ext, 0) : KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+        rank > 0 ? mapping.stride(0) : 0,
+        rank > 1 ? dimension_from_extent(ext, 1) : KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+        rank > 1 ? mapping.stride(1) : 0,
+        rank > 2 ? dimension_from_extent(ext, 2) : KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+        rank > 2 ? mapping.stride(2) : 0,
+        rank > 3 ? dimension_from_extent(ext, 3) : KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+        rank > 3 ? mapping.stride(3) : 0,
+        rank > 4 ? dimension_from_extent(ext, 4) : KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+        rank > 4 ? mapping.stride(4) : 0,
+        rank > 5 ? dimension_from_extent(ext, 5) : KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+        rank > 5 ? mapping.stride(5) : 0,
+        rank > 6 ? dimension_from_extent(ext, 6) : KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+        rank > 6 ? mapping.stride(6) : 0,
+        rank > 7 ? dimension_from_extent(ext, 7) : KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+        rank > 7 ? mapping.stride(7) : 0,
+    };
+  } else {
+    return ArrayLayout{
+        rank > 0 ? dimension_from_extent(ext, 0) : KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+        rank > 1 ? dimension_from_extent(ext, 1) : KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+        rank > 2 ? dimension_from_extent(ext, 2) : KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+        rank > 3 ? dimension_from_extent(ext, 3) : KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+        rank > 4 ? dimension_from_extent(ext, 4) : KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+        rank > 5 ? dimension_from_extent(ext, 5) : KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+        rank > 6 ? dimension_from_extent(ext, 6) : KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+        rank > 7 ? dimension_from_extent(ext, 7)
+                 : KOKKOS_IMPL_CTOR_DEFAULT_ARG};
+  }
+}
+
+template <class MDSpanType, class VM>
+KOKKOS_INLINE_FUNCTION auto mapping_from_view_mapping(const VM &view_mapping) {
+  using mapping_type = typename MDSpanType::mapping_type;
+  using extents_type = typename mapping_type::extents_type;
+
+  if constexpr (std::is_same_v<typename mapping_type::layout_type,
+                               Kokkos::layout_stride>) {
+    // Workaround for no span in 17...
```

Comment:
I find this comment a bit cryptic.
What about something like
```suggestion
    // Using std::array instead of std::span here because au current minimum requirement is C++17 and std::span was added in C++20.  Revisit when C++20 is available.
```
(not formatted)

Reply:
Done